src/duckling/canvas/tools/entity-move-tool.ts

```diff
@@ -52,13 +54,18 @@ export class EntityMoveTool extends BaseTool {
 
     onStageMove(event : CanvasMouseEvent) {
         if (this._selection) {
-            this._entityPositionService.setPosition(
-                this._selection,
-                vectorRound(vectorAdd(event.stageCoords, this._selectOffsetCoords)),
-                this._mergeKey);
+            let destination = vectorRound(vectorAdd(event.stageCoords, this._selectOffsetCoords));
+            let gridSize = this._projectService.project.getValue().currentMap.gridSize;
+            if (!this._isSnapToGrid(event)) {
+                let gridSize = this._projectService.project.getValue().currentMap.gridSize;
```

Comment:
This is a duplicate.
let gridSize = this._projectService.project.getValue().currentMap.gridSize;